airbyte-persistence/job-persistence/src/main/java/io/airbyte/persistence/job/factory/OAuthConfigSupplier.java

```diff
@@ -47,6 +49,36 @@ public static boolean hasOAuthConfigSpecification(final ConnectorSpecification s
     return spec != null && spec.getAdvancedAuth() != null && spec.getAdvancedAuth().getOauthConfigSpecification() != null;
   }
 
+  public JsonNode maskSourceOAuthParameters(final UUID sourceDefinitionId, final UUID workspaceId, final JsonNode sourceConnectorConfig)
```

Comment:
Added some new methods to _mask_ oauth parameters from a provided config and ensure we don't write our global oauth params to the connector-specific config / secrets.